### ISSUE TITLE
bug: Tricky bug with listchar 

### ISSUE THREAD

**Issue by titouancreach** [stevearc/oil.nvim#177]
### Did you check the docs and existing issues?

- [X] I have read the docs
- [X] I have searched the existing issues

### Neovim version (nvim -v)

NVIM v0.9.1

### Operating system/version

MAC OS 13.4 Beta (22F5027f)

### Describe the bug

I've a setup where I use `NvimTree` + `oil` (`NvimTree` to have a big picture of the project tree) and `Oil` for the mutations.
I also have this in my config that acts as a bridge between `NvimTree` and `Oil`

```lua
    {
        "FileType",
        {
            pattern = { "NvimTree" },
            callback = function()
                vim.keymap.set('n', '<leader>o', function()
                    local api = require('nvim-tree.api')
                    local node = api.tree.get_node_under_cursor()
                    local path = node.absolute_path

                    if (path and vim.fn.isdirectory(path) == 0) then
                        path = vim.fn.fnamemodify(path, ':h')
                    end

                    require("oil").open(path)
                end, { buffer = true })
            end
        }
    },
```

An Autocmd that add a binding to open `Oil` at the right plave when I'm naviguating in `NvimTree`. And also, I use "set list" and `listchars` to show invisibles chars.

Now the fun part:

If open NvimTree, then press \<leader\>o, then navigate to a file using the oil buffer, then open it with \<CR\>, no more listchar.
(Yes it's tricky, this is the first time I can repro 100% this is why I open an issue now)

I believe this is because Oil want to restore my settings and, I come from another special buffer (NvimTree) that has probably "set nolist" before (probably here or something: https://github.com/nvim-tree/nvim-tree.lua/blob/94c7c810af205c0f00c8f105dcf490c8eb17658a/lua/nvim-tree/view.lua#L28)

To be honest, I don't know how we can fix this, and I don't know if the issue should be in this repo or in my own config.

I just think Oil should not change my settings in all my buffers.

### Steps To Reproduce

1. nvim -u repro.lua repro.lua (you can see list char)
2. space e (to jump to lvim tree)
3. move cursor on repro.lua and press space o to open oil
4. press \<CR\>
5. no more list char

### Expected Behavior

should see listchar

### Repro

```Lua
vim.keymap.set("n", " ", "<Nop>", { silent = true, remap = false })
vim.g.mapleader = " "
-- save as repro.lua
-- run with nvim -u repro.lua
-- DO NOT change the paths
local root = vim.fn.fnamemodify("./.repro", ":p")

-- set stdpaths to use .repro
for _, name in ipairs({ "config", "data", "state", "runtime", "cache" }) do
    vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

-- bootstrap lazy
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
    vim.fn.system({
        "git",
        "clone",
        "--filter=blob:none",
        "--single-branch",
        "https://github.com/folke/lazy.nvim.git",
        lazypath,
    })
end
vim.opt.runtimepath:prepend(lazypath)

-- install plugins
local plugins = {
    "folke/tokyonight.nvim",
    {
        "stevearc/oil.nvim",
        config = function()
            require("oil").setup({
                -- add any needed settings here
            })
        end,
    },
    {
        "nvim-tree/nvim-tree.lua",
        version = "*",
        lazy = false,
        dependencies = {
            "nvim-tree/nvim-web-devicons",
        },
        config = function()
            require("nvim-tree").setup {}
        end,
    }
    -- add any other plugins here
}
require("lazy").setup(plugins, {
    root = root .. "/plugins",
})


vim.opt.listchars = { eol = '↵', space = '·', tab = '>-', nbsp = '␣' }
vim.opt.list = true
ƒ

vim.api.nvim_create_autocmd("FileType", {
    pattern = { "NvimTree" },
    callback = function()
        vim.keymap.set('n', '<leader>o', function()
            local api = require('nvim-tree.api')
            local node = api.tree.get_node_under_cursor()
            local path = node.absolute_path

            if (path and vim.fn.isdirectory(path) == 0) then
                path = vim.fn.fnamemodify(path, ':h')
            end

            require("oil").open(path)
        end, { buffer = true })
    end
})

vim.keymap.set("n", '<leader>e', '<cmd>NvimTreeOpen<CR>', {});

vim.cmd.colorscheme("tokyonight")
-- add anything else here
```

### Did you check the bug with a clean config?

- [X] I have confirmed that the bug reproduces with `nvim -u repro.lua` using the repro.lua file above.

**Comment by stevearc** [comment]
Thanks for reporting this! This prodded me to dig a little deeper into the root causes, and it turns out the oil logic for restoring window options isn't needed anymore. 9e036c6a4868b971127f3cb6bac6197bb4103723
So I was able to remove some dead code, get rid of a config option, and also hopefully fix your issue!

**Comment by titouancreach** [comment]
I can confirm this is fixed with 9e036c6a4868b971127f3cb6bac6197bb4103723. Thanks for your reactiviy !
